### PR TITLE
Document 'Delete' reclaim policy

### PR DESCRIPTION
**Pull request by jsafrane**

### CODE REVIEW

File: docs/user-guide/persistent-volumes/index.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ When a user is done with their volume, they can delete the PVC objects from the
 
 ### Reclaiming
 
-The reclaim policy for a `PersistentVolume` tells the cluster what to do with the volume after it has been released.  Currently, volumes can either be Retained or Recycled.  Retention allows for manual reclamation of the resource.  For those volume plugins that support it, recycling performs a basic scrub (`rm -rf /thevolume/*`) on the volume and makes it available again for a new claim.
+The reclaim policy for a `PersistentVolume` tells the cluster what to do with the volume after it has been released.  Currently, volumes can either be Retained, Recycled or Deleted.  Retention allows for manual reclamation of the resource.  For those volume plugins that support it, deletion removes both the `PersistentVolume` object from Kubernetes as well as associated storage asset in cloud infrastructure and recycling performs a basic scrub (`rm -rf /thevolume/*`) on the volume and makes it available again for a new claim.
```
Comment (markturansky) [inline docs/user-guide/persistent-volumes/index.md]: "...has been released of its claim."

Comment (markturansky) [inline docs/user-guide/persistent-volumes/index.md]: "... asset in cloud infrastructure.  Recycling performs..."

File: docs/user-guide/persistent-volumes/index.md [hunk 2]
```diff
@@ -109,8 +109,9 @@ Current recycling policies are:
 
 * Retain -- manual reclamation
 * Recycle -- basic scrub ("rm -rf /thevolume/*")
+* Delete -- associated storage asset such as AWS EBS, GCE PD or OpenStack Cinder volume is deleted
 
-Currently, NFS and HostPath support recycling.
+Currently, only NFS and HostPath support recycling and AWS EBS, GCE PD and Cinder volumes support deletion.
```
Comment (markturansky) [inline docs/user-guide/persistent-volumes/index.md]: "...support recycling.  AWS EBS, GCE..."

**Comment by markturansky** [comment]
Couple of nits to shorten sentences but the content is perfect.

File: docs/user-guide/persistent-volumes/index.md (hunk 1 above)
Comment (jsafrane) [inline docs/user-guide/persistent-volumes/index.md]: > "... asset in cloud infrastructure. Recycling performs..."

The "For those volume plugins that support" part should clearly apply to both recycling and deletion. I split it into two sentences, both explicitly mentioning plugin support.

File: docs/user-guide/persistent-volumes/index.md (hunk 2 above)
Comment (jsafrane) [inline docs/user-guide/persistent-volumes/index.md]: Ok

**Comment by jsafrane** [comment]
Updated and rephrased a bit.

File: docs/user-guide/persistent-volumes/index.md (hunk 1 above)
Comment (markturansky) [inline docs/user-guide/persistent-volumes/index.md]: Good enough for me :)

**Comment by johndmulhausen** [comment]
Thanks for the review, @markturansky and thanks for the PR @jsafrane !

**Pull request merged**